log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationSource.java

```diff
@@ -57,8 +63,8 @@ public class ConfigurationSource {
             new ConfigurationSource(Constants.EMPTY_BYTE_ARRAY, null, 0);
 
     private final InputStream stream;
-    private volatile byte[] data;
-    private volatile Source source;
+    private volatile byte /*@Nullable*/[] data;
+    private final /*@Nullable*/ Source source;
     private final long lastModified;
     // Set when the configuration has been updated so reset can use it for the next lastModified timestamp.
     private volatile long modifiedMillis;
```

Comment:
Could you document `lastModified` and `modifiedMillis` class fields, please? As a matter of fact, I find the names very confusing too. I would really appreciate an improvement and uniformity there too. For instance, does `lastModified` also contain millis? If so, why isn't it prefixed with `Millis`? etc. In short, please rename these two fields too.

Reply:
Done in https://github.com/apache/logging-log4j2/pull/2941/commits/3598b6a863dfd6028e72c137936e490ef695a791